actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -507,7 +507,7 @@ def masked_authenticity_token(form_options: {})
       # Checks the client's masked token to see if it matches the session token.
       # Essentially the inverse of `masked_authenticity_token`.
       def valid_authenticity_token?(session, encoded_masked_token) # :doc:
-        if encoded_masked_token.nil? || encoded_masked_token.empty? || !encoded_masked_token.is_a?(String)
```

Comment:
Hum, it's curious that was a check for it already.

Reply:
This was added in https://github.com/rails/rails/commit/d0303d03a94994b7653c629f3cf1578c82a3eccf by @rafaelfranca but as some sort of followup on https://github.com/rails/rails/pull/18917 which also assume the authenticity token may not be a string.

I still don't get how it could legitimately not be a string. 

But since the  check is already here anyway, reordering it so it's more robust makes sense to me.

Reply:
We had an issue with penetration testing where the client was forcing a numeric value through the headers. In real life it would be a string, but some clients appear to force it as a plain value.

Reply:
Yes, using a JSON request, it's possible to pass an integer.